### PR TITLE
Add support for opacity text animators

### PR DESCRIPTION
**Pull request by calda**
This PR adds basic support for opacity text animators in the main thread rendering engine:

### CODE REVIEW

File: Sources/Private/MainThread/LayerContainers/CompLayers/PreCompositionLayer.swift [hunk 1]
```diff
@@ -16,6 +16,7 @@ final class PreCompositionLayer: CompositionLayer {
     asset: PrecompAsset,
     layerImageProvider: LayerImageProvider,
     layerTextProvider: LayerTextProvider,
+    layerFontProvider: LayerFontProvider,
```
Comment (1an-xu) [inline Sources/Private/MainThread/LayerContainers/CompLayers/PreCompositionLayer.swift:19]: rebase

Comment (calda) [reply Sources/Private/MainThread/LayerContainers/CompLayers/PreCompositionLayer.swift]: thx, fixed it

File: Sources/Private/Model/Text/TextAnimator.swift [hunk 2]
```diff
@@ -164,7 +209,7 @@ final class TextAnimator: Codable, DictionaryInitializable {
   // MARK: Private
 
   private enum CodingKeys: String, CodingKey {
-//    case textSelector = "s" TODO
```
Comment (1an-xu) [inline Sources/Private/Model/Text/TextAnimator.swift:167]: remove

**Review by 1an-xu — APPROVED** [review]

**Comment by alisamili** [comment]
> 此 PR 在主线程渲染引擎中添加了对不透明文本动画器的基本支持：
> 
>  [  ](https://private-user-images.githubusercontent.com/1811727/340405096-f0973b0c-d9cd-4a69-a2b3-5ef7c72af815.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMTMyNjEsIm5iZiI6MTcyNTAxMjk2MSwicGF0aCI6Ii8xODExNzI3LzM0MDQwNTA5Ni1mMDk3M2IwYy1kOWNkLTRhNjktYTJiMy01ZWY3YzcyYWY4MTUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzBUMTAxNjAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzkwMjlkYWJjYzFhZDY1YmU2ZGM3OTk4MGY4ODA3ZDM3MmNlNWVkMzIzMjFhYTUwMzRlODc3NjczYzFmODIxZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.oMMrmEGVeCrRIZvMA9N3DIbdSuYRm7jufPn_ZDs5hBk) [ ](https://private-user-images.githubusercontent.com/1811727/340405096-f0973b0c-d9cd-4a69-a2b3-5ef7c72af815.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMTMyNjEsIm5iZiI6MTcyNTAxMjk2MSwicGF0aCI6Ii8xODExNzI3LzM0MDQwNTA5Ni1mMDk3M2IwYy1kOWNkLTRhNjktYTJiMy01ZWY3YzcyYWY4MTUuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzBUMTAxNjAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MzkwMjlkYWJjYzFhZDY1YmU2ZGM3OTk4MGY4ODA3ZDM3MmNlNWVkMzIzMjFhYTUwMzRlODc3NjczYzFmODIxZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.oMMrmEGVeCrRIZvMA9N3DIbdSuYRm7jufPn_ZDs5hBk)
> 
>  [  ](https://private-user-images.githubusercontent.com/1811727/340405025-b5016edd-f485-4411-b912-1ac5f9df8f4d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMTMyNjEsIm5iZiI6MTcyNTAxMjk2MSwicGF0aCI6Ii8xODExNzI3LzM0MDQwNTAyNS1iNTAxNmVkZC1mNDg1LTQ0MTEtYjkxMi0xYWM1ZjlkZjhmNGQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzBUMTAxNjAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjFjNjdiZTlhOGU5Y2FhYjIzNDVmMWNkZDNjZmM2MWI5NzgwODNlODMwYzM2ZDAxZjIyYWVmYjM3NzE3MDg5OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.XF8WFZ0-X77i-dKVJ5VqLuScmcUd5SaVRsTS0x3NOAM) [ ](https://private-user-images.githubusercontent.com/1811727/340405025-b5016edd-f485-4411-b912-1ac5f9df8f4d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwMTMyNjEsIm5iZiI6MTcyNTAxMjk2MSwicGF0aCI6Ii8xODExNzI3LzM0MDQwNTAyNS1iNTAxNmVkZC1mNDg1LTQ0MTEtYjkxMi0xYWM1ZjlkZjhmNGQuZ2lmP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzBUMTAxNjAxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NjFjNjdiZTlhOGU5Y2FhYjIzNDVmMWNkZDNjZmM2MWI5NzgwODNlODMwYzM2ZDAxZjIyYWVmYjM3NzE3MDg5OCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.XF8WFZ0-X77i-dKVJ5VqLuScmcUd5SaVRsTS0x3NOAM)


hi，can u help me？
let replacements = ["#textnodea": getStringWithKey("KEY_xxxxa"), "#textnodeb": getStringWithKey("KEY_xxxxb")]
In 3.3.0, using DictionaryTextProvider (replacements), replace the internationalized multi-language key. In countries where Arabic is used, the text content is still displayed in English. Can this version solve this problem?

**Pull request merged**